vg/draw/canvas.go

```diff
@@ -264,36 +283,22 @@ func New(c vg.CanvasSizer) Canvas {
 }
 
 // NewFormattedCanvas creates a new vg.CanvasWriterTo with the specified
-// image format.
-//
-// Supported formats are:
+// image format. Supported formats need to be registered by importing one or
+// more of the following packages:
 //
-//  eps, jpg|jpeg, pdf, png, svg, and tif|tiff.
+//     gonum.org/v1/plot/vg/vgimg // provides png, jpg|jpeg, tif|tiff
+//     gonum.org/v1/plot/vg/vgsvg // provides svg
+//     gonum.org/v1/plot/vg/vgeps // provides eps
+//     gonum.org/v1/plot/vg/vgpdf // provides pdf
 func NewFormattedCanvas(w, h vg.Length, format string) (vg.CanvasWriterTo, error) {
-	var c vg.CanvasWriterTo
-	switch format {
-	case "eps":
-		c = vgeps.New(w, h)
-
-	case "jpg", "jpeg":
-		c = vgimg.JpegCanvas{Canvas: vgimg.New(w, h)}
-
-	case "pdf":
-		c = vgpdf.New(w, h)
-
-	case "png":
-		c = vgimg.PngCanvas{Canvas: vgimg.New(w, h)}
-
-	case "svg":
-		c = vgsvg.New(w, h)
-
-	case "tif", "tiff":
-		c = vgimg.TiffCanvas{Canvas: vgimg.New(w, h)}
-
-	default:
-		return nil, fmt.Errorf("unsupported format: %q", format)
+	formats.Lock()
+	defer formats.Unlock()
+	for name, fn := range formats.m {
+		if format == name {
+			return fn(w, h), nil
+		}
 	}
-	return c, nil
+	return nil, fmt.Errorf("unsupported format: %q", format)
```

Comment:
perhaps also print the known, supported formats?

Reply:
Done.